resources/templates/md/posts/2016-04-04-state.md

```diff
@@ -0,0 +1,366 @@
+{:title "The approach to state management in Clojure"
+:description  "Intro to mutable state in Clojure. Greg the llama show you an effective way how to manage state, and you'll find out that llamas love candies too :)"
+:layout :post
+:tags  ["Clojure" "ClojureScript" "Fundamentals"]
+:toc false
+:author "Dajana Štiberová"
+:image ""}
+
+Long time ago, during one interview, I was asked to describe the difference between `vars` and `atoms`.
+It was a pretty simple question, but at that time it was not so easy to describe this difference for me.
+Now I'm more skilled in Clojure but I still think that it was an interesting question and when
+I was looking for a topic for the blog post, I decided to use this one.
+
+So if you want to read something about state management, or you want to know how to dance like Greg the llama, this post is for you :)
+
+## States and Identities
+
+We need to know that in Clojure there is a difference between state and identity.
+
+`States` are an immutable data.
+A state is the value an identity points to at a particular time. State never changes.
+
+`Identities` unify different values over the time.
+Identity types are mutable references to immutable values,
+and can be modelled by using one of the four references types (`Var`s, `Atom`s, `Agent`s, `Ref`s)
+
+
+### *Example*
+
+Our example is the best llama on earth, Greg the llama.
+Greg is an identity.
+Greg wants to go to the party with Gregorina. He is a good dancer, and he wants to teach us how to dance.
+
+So, Greg is dancing and every snapshot of this action is one state.
```

Comment:
I would maybe highlight *snapshot*, to make it clear that the point is "pictures at a given time"